update version number in spec and references to spec (#607)
eclipse/microprofile-open-api: spec/src/main/asciidoc/microprofile-openapi-spec.asciidoc

the MP Config specification.

There are various ways to inject these configuration values into an MP OpenAPI
framework, including the https://github.com/eclipse/microprofile-config/blob/main/spec/src/main/asciidoc/configsources.asciidoc#default-configources[default ConfigSource] as well as

> https://github.com/eclipse/microprofile-config/blob/main/spec/src/main/asciidoc/configsources.asciidoc#custom-configsources[custom ConfigSource].

Reviewer: Link to the MP config spec, not the source code: https://download.eclipse.org/microprofile/microprofile-config-3.1/microprofile-config-spec-3.1.html

|===
| Annotation | Description

> | https://github.com/eclipse/microprofile-open-api/blob/main/api/src/main/java/org/eclipse/microprofile/openapi/annotations/callbacks/Callback.java[@Callback] | Represents a callback URL that will be invoked.

Reviewer: I don't really like linking into the source like this, particularly linking to the `main` branch, since that might be re-factored at any time.

Ideally we would link to the javadoc, though then we need to generate spec based on where we expect the javadoc to be when the spec is published. If we do link to the source then we should link to the tag, which runs into the same issue (the tag doesn't exist yet).

I'll raise another issue to improve this (though I'm not sure it'll get done before release).

Reviewer: Raised #609 to cover removing links to source code from the spec.

| https://github.com/eclipse/microprofile-open-api/blob/main/api/src/main/java/org/eclipse/microprofile/openapi/annotations/media/Content.java[@Content] | Provides schema and examples for a particular media type.
| https://github.com/eclipse/microprofile-open-api/blob/main/api/src/main/java/org/eclipse/microprofile/openapi/annotations/media/DiscriminatorMapping.java[@DiscriminatorMapping] | Used to differentiate between other schemas which may satisfy the payload description.
| https://github.com/eclipse/microprofile-open-api/blob/main/api/src/main/java/org/eclipse/microprofile/openapi/annotations/media/Encoding.java[@Encoding] | Single encoding definition to be applied to single Schema Object.
| https://github.com/eclipse/microprofile-open-api/blob/main/api/src/main/java/org/eclipse/microprofile/openapi/annotations/media/ExampleObject.java[@ExampleObject] | Illustrates an example of a particular content.
| https://github.com/eclipse/microprofile-open-api/blob/main/api/src/main/java/org/eclipse/microprofile/openapi/annotations/media/Schema.java[@Schema] | Allows the definition of input and output data types.

> | https://github.com/eclipse/microprofile-open-api/blob/main/api/src/main/java/org/eclipse/microprofile/openapi/annotations/media/Schema.java[@SchemaProperty] | Allows the definition of a property nested within a parent @Schema.

Reviewer: ```suggestion
| https://github.com/eclipse/microprofile-open-api/blob/main/api/src/main/java/org/eclipse/microprofile/openapi/annotations/media/SchemaProperty.java[@SchemaProperty] | Allows the definition of a property nested within a parent @Schema.
```

Reviewer: This was wrong before.

the default `YAML` format and `JSON` format.

=== Context root behavior

> Vendors are required to ensure that the combination of each global https://spec.openapis.org/oas/v3.1.0.html#server-object[server]
> element and https://spec.openapis.org/oas/v3.1.0.html#path-item-object[path-item] element resolve to the absolute backend URL of that
> particular path.  If that `pathItem` contains a `servers` element , then this
> list of operation-level `server` elements replaces the global list of servers
> for that particular `pathItem`.

Reviewer: ```suggestion
Vendors are required to ensure that the combination of each global https://spec.openapis.org/oas/v3.1.0.html#server-object[server]
element and https://spec.openapis.org/oas/v3.1.0.html#path-item-object[pathItem] element resolve to the absolute backend URL of that
particular path.  If that `pathItem` contains a `servers` element , then this
list of operation-level `server` elements replaces the global list of servers
for that particular `pathItem`.
```

to make outbound calls into another microservice (B).  In this case, we do not want
the interface for microservice (B) to appear in microservice (A)'s OAS3 document.

> Therefore, vendors are required to exclude from the final OAS3 document any interface annotated with https://github.com/eclipse/microprofile-rest-client/blob/main/api/src/main/java/org/eclipse/microprofile/rest/client/inject/RegisterRestClient.java[org.eclipse.microprofile.rest.client.inject.RegisterRestClient].

Reviewer: ```suggestion
Therefore, vendors are required to exclude from the final OAS3 document any interface annotated with https://download.eclipse.org/microprofile/microprofile-rest-client-3.0/apidocs/org/eclipse/microprofile/rest/client/inject/RegisterRestClient.html[org.eclipse.microprofile.rest.client.inject.RegisterRestClient].
```
